London | May-2025 | Halimatou Saddiyaa | Form controls

Halimatou-saddiyaa opened the pull request:
<!--

You must title your PR like this:

REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME

For example,

London | May-2025 | Carol Owen | Wireframe

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

Please review my Form Controls PR, thanks.

## Questions

Ask any questions you have for your reviewer.

day-lee commented on Form-Controls/index.html:
Hi Halimatou, I noticed that your form isn't working as intended. Have you tested it with a submit button? You can add a button inside the form and try entering some input to test its functionality.

day-lee commented on Form-Controls/index.html:
```diff
@@ -12,15 +12,65 @@
+          <input type="email" id="mail" name="user_email" />
+        </p>
+        <p>
+          <label for="t1"
```
label is used for accessibility. it needs a clearer name to screen reader can give better information. 

FYI: The for attribute in a <label> tag should reference a meaningful and descriptive ID. This improves accessibility by helping screen readers provide clearer and more helpful information to users. A clear and specific for value ensures that assistive technologies can accurately describe the input's purpose.

cjyuan commented:
@Halimatou-saddiyaa 

In CYF courses, the recommended way to inform the reviewer of your changes is to do both of the following:
  - Reply to their feedback. (Can you respond to each of @day-lee feedback?)
    - In the responses, clarify how each piece of feedback was addressed to demonstrate that you've carefully reviewed the suggestions.
    - Your response may trigger a notification (depending on the reviewer's settings), helping ensure they’re aware of the updates you’ve made. 
  - Replace the "Reviewed" label by a "Needs review" label (which you have done -- great!)

Halimatou-saddiyaa commented:
Thanks for your comment @day-lee . 
I've changed the description of the tags "label" to be more informative and give more information for the screen reader.

Halimatou-saddiyaa commented:
Hi @day-lee,
 I've added a submit button, and changed my code to improve the functionnality of my form.
I've tested it and now the form is working as intended, I think.
Can you please review again?
Thanks

Halimatou-saddiyaa commented:
@cjyuan I've done that when I first received the feedback but it seems like it did not trigger a notification to the reviewer.
![Form controls PR review](https://github.com/user-attachments/assets/23f68437-8ba2-49d6-9993-61622a676f65)
![Form controls PR review 2](https://github.com/user-attachments/assets/0db8f05c-29a4-4b63-a6fe-e63ff2321f7a)

cjyuan commented on Form-Controls/index.html:
```diff
@@ -8,20 +8,46 @@
+      </p>
+      <p>
+        <label for="size">Size of tee-shirt?<span aria-label="required">*</span></label>
+        <input type="text" id="size" name="size" list="sizes" required />
+        <datalist id="sizes">
+          <option>XS</option>
+          <option>S</option>
+          <option>M</option>
+          <option>L</option>
+          <option>XL</option>
+          <option>XXL</option>
+        </datalist>
```
With this kind of input element, the user can input a value not listed as one of the options specified in `<datalist>`. 

Can you look up what HTML element you need to enforce "select one of the 6 sizes"? (Ask ChatGPT)

cjyuan commented on Form-Controls/index.html:
```diff
@@ -8,20 +8,46 @@
+        <input type="email" id="mail" name="user_email" />
+      </p>
+      <p>
+        <label for="colour">Colour of tee-shirt?<span aria-label="required">*</span></label>
```
What is the meaning of `*` in the label? Is it necessary?

cjyuan reviewed (commented):
@day-lee Since the deadline to enroll as a trainee is 6/6, I’m jumping in to continue the review in this PR.

@Halimatou-saddiyaa 
1. According to https://validator.w3.org/, there are no errors but there are some warnings.  Can you fix the warnngs?

2. Your most recent commit on this branch was made on 20 May. Is it your latest change?

3. I was able to submit the form without providing any name or email. (This is against the specification).

4. I saw some very good suggestions when I asked ChatGPT this:
```
review HTML code:
...        <--- paste your HTML code here
```

To get better at using AI tools, try running your code by ChatGPT. Take a look at the suggestions it gives you, and pick out the ones that actually make sense for your project. It's a great way to learn and improve your work.

cjyuan commented:
Apparently `<div>` does not produce the same spacing `<p>` does without CSS. So to reach 100 in Lighthouse Accessibility score, you will need to add some `<br>` to create vertical space between the input elements. I am sure you can change it with ease.

I will go ahead and mark this PR as "Complete".

In addition to replacing labels, please also remember to respond to reviewer's comments in future PRs.
